src/pages/index.js

```diff
@@ -1,84 +1,212 @@
 import Card from "../components/scripts/Card.js";
 import FormValidation from "../components/scripts/FormValidator.js";
-// import Popup from "../scripts/Popup.js";
 import PopupWithForm from "../components/scripts/PopupWithForm.js";
 import PopupWithImage from "../components/scripts/PopupWithImage.js";
 import UserInfo from "../components/scripts/UserInfo.js";
 import Section from "../components/scripts/Section.js";
 import * as DOM from "../utils/constants.js";
 import { validationSettings } from "../utils/utils.js";
-import { initialCards } from "../utils/utils.js";
-
+import Api from "../components/scripts/Api.js";
+import PopupWithConfirmation from "../components/scripts/PopupWithConfirmation.js";
 import "../pages/index.css";
 
+// Api
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "d081ca18-dd96-4ff1-863a-c195fbcda74a",
+    "Content-Type": "application/json",
+  },
+});
+
+const userInformation = new UserInfo(
+  ".profile__title",
+  ".profile__description",
+  ".profile__pic"
+);
+
+const formValidators = {};
+const enableValidation = (validationSettings) => {
+  DOM.forms.forEach((formElement) => {
+    const validator = new FormValidation(validationSettings, formElement);
+    const formName = formElement.getAttribute("name");
+    formValidators[formName] = validator;
+    validator.enableValidation();
+  });
+};
+
+enableValidation(validationSettings);
+
+let section;
+
+Promise.all([api.getInitialCards(), api.getUserInfo()])
+  .then(([cardData, formData]) => {
+    section = new Section(
+      {
+        items: cardData,
+        renderer: (item) => {
+          const cardElement = createCard(item);
+          section.addItem(cardElement);
+        },
+      },
+      ".cards__list"
+    );
+    section.renderItems();
+
+    userInformation.setUserInfo(formData);
+    userInformation.setAvatar(formData.avatar);
+  })
+  .catch((err) => {
+    console.error(err);
+  });
+
+function handleDeleteCardClick(item) {
+  popupWithConfirmation.open();
+  popupWithConfirmation.setSubmitCall(() => {
+    popupWithConfirmation.deleting(true);
+    api
+      .deleteCard(item.getId())
+      .then(() => {
+        item.handleDeleteCard();
+        popupWithConfirmation.deleting(false);
+      })
+      .then(popupWithConfirmation.close())
```

Comment:
`      .then(popupWithConfirmation.close())
`

In blocks `then`, `catch`, `finally` you need to pass only the **reference** to the function (not a function call) otherwise the function is called immediately without waiting for the response from the server. You can pass an **unnamed** arrow function as well. 

**Please, correct it everywhere**